js/packages/teams-ai/src/authentication/BotAuthentication.ts

```diff
@@ -0,0 +1,280 @@
+    ): void {
+        this._userSignInFailureHandler = handler;
+    }
+
+    public async handleSignInActivity(context: TurnContext, state: TState): Promise<void> {
+        try {
+            const userDialogStatePropertyName = this.getUserDialogStatePropertyName(context);
+            const result = await this.runDialog(context, state, userDialogStatePropertyName);
+
+            if (result.status === DialogTurnStatus.complete) {
+                // OAuthPrompt dialog should have sent an invoke response already.
+
+                if (result.result?.token) {
+                    // Successful sign in
+                    setTokenInState(state, this._settingName, result.result.token);
+
+                    // Get user auth state
+                    const userAuthState = state.conversation.value[
+                        this.getUserAuthStatePropertyName(context)
+                    ] as UserAuthState;
+
+                    // Restore previous user message
+                    context.activity.text = userAuthState.message || '';
+
+                    await this._userSignInSuccessHandler?.(context, state);
+                } else {
+                    // Failed sign in
+                    await this._userSignInFailureHandler?.(
+                        context,
+                        state,
+                        new AuthError('Authentication flow completed without a token.', 'completionWithoutToken')
+                    );
+                }
+            }
+        } catch (e) {
+            const errorMessage = e instanceof Error ? e.message : JSON.stringify(e);
+            const message = `Unexpected error encountered while signing in: ${errorMessage}. 
+                Incomming activity details: type: ${context.activity.type}, name: ${context.activity.name}`;
+
+            await this._userSignInFailureHandler?.(context, state, new AuthError(message));
+        }
+    }
+
+    public async runDialog(
+        context: TurnContext,
+        state: TState,
+        dialogStateProperty: string
+    ): Promise<DialogTurnResult<OAuthPromptResult>> {
+        const accessor = new TurnStateProperty<DialogState>(state, 'conversation', dialogStateProperty);
+        const dialogSet = new DialogSet(accessor);
+        dialogSet.add(this._oauthPrompt);
+        const dialogContext = await dialogSet.createContext(context);
+        let results = await dialogContext.continueDialog();
+        if (results.status === DialogTurnStatus.empty) {
+            results = await dialogContext.beginDialog(this._oauthPrompt.id);
+        }
+        return results;
+    }
+
+    public deleteAuthFlowState(context: TurnContext, state: TState) {
+        // Delete user auth state
+        const userAuthStatePropertyName = this.getUserAuthStatePropertyName(context);
+        if (state.conversation.value[userAuthStatePropertyName]) {
+            delete state.conversation.value[userAuthStatePropertyName];
```

Comment:
Please take note: I think Steve merged in his latest PR that gets rid of the need to add `value` - so you might want to verify before this PR gets checked in.

Reply:
Yeah i'll have to merge his changes in.